pallets/attesters/contracts/contracts/remoteOrder.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract RemoteOrder {
+
+    event OrderCreated(bytes32 indexed id, bytes4 indexed destination, bytes4 asset, address targetAccount, uint256 amount, address rewardAsset, uint256 insurance, uint256 maxReward, uint32 nonce);
+    event OrderCommitted(bytes32 indexed id);
+    event OrderReverted(bytes32 indexed id);
+    event OrderRefundedInERC20(bytes32 indexed id, address indexed asset, uint256 amount);
+
+    struct Order {
+        bool exists;
+        bytes4 destination;
+        bytes4 asset;
+        address targetAccount;
+        address sender;
+        uint256 amount;
+        Reward reward;
+        OrderStatus status;
+    }
+
+    struct Reward {
+        address asset;
+        uint256 insurance;
+        uint256 maxReward;
+    }
+
+    enum OrderStatus {
+        Accepted,
+        Committed,
+        Reverted
+    }
+
+    mapping(address => uint32) public requestNonce;
+    mapping(bytes32 => Order) public orders;
+    address owner;
+
+    constructor() {
+        owner = msg.sender;
+    }
+
+    function generateId(address requester, uint32 nonce) public pure returns (bytes32) {
+        return keccak256(abi.encode(requester, nonce));
+    }
+
+    /*
+     * Before making the order, the function checks that the user has enough balance (of either Ether or the ERC20 token).
+     * If everything is okay, it increases the nonce for the user, creates a unique id for the order, saves the order in the mapping,
+     * and emits the OrderCreated event.
+     */
+    function order(
+        bytes4 destination,
+        bytes4 asset,
+        address targetAccount,
+        uint256 amount,
+        address rewardAsset,
+        uint256 insurance,
+        uint256 maxReward
+    ) public payable {
+        require(maxReward > 0, "Max reward must be greater than 0");
+
+        uint32 nonce = requestNonce[msg.sender];
+        requestNonce[msg.sender] = nonce + 1;
```

Comment:
I have security concerns about this. nonce + 1 seems like it can overflow
not sure about severity tho